jobs/cache_maintenance/src/cache_maintenance/delete_indexes.py

```diff
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import glob
+import logging
+import os
+from datetime import datetime, timedelta
+
+from libcommon.storage import StrPath
+
+
+def delete_indexes(
+    duckdb_index_cache_directory: StrPath, subdirectory: str, file_extension: str, expired_time_interval_seconds: int
+) -> None:
```

Comment:
it deletes the files but not the directories, ie currently, we have on the disk: `downloads/asoria-copy_final-e388ab6d/asoria--copy_final/train/index.duckdb`. It's the only file in `downloads/asoria-copy_final-e388ab6d/`. Shouldn't we try to delete the directories too? On the other side: it might lead to concurrency issues (/search creates the file, while the job deletes the directories). Not sure how to manage this.

---

Not related to this particular PR, but we have a somewhat similar situation for the temporal `duckdb-index` directories: many empty directories. But in that case, they should have been used only once. So these should be deleted.

Reply:
I think I could change the logic for index downloads, instead of downloading in separate folders dataset/config/split, maybe they can be a combination in the index name as `dataset-config-split-revision-index.duckdb`  so that, this job only has to remove files without dealing with empty directories.

Reply:
yes, it seems better!

Reply:
So: in another PR